applications/StructuralMechanicsApplication/custom_utilities/constitutive_law_utilities.cpp

```diff
@@ -942,6 +942,222 @@ Matrix ConstitutiveLawUtilities<TVoigtSize>::CalculateDirectPlasticDeformationGr
+        rAs(5, 5) = r_mat_props[ISOTROPIC_ANISOTROPIC_YIELD_RATIO_XZ];        
+    } else {
+        rAs(0, 0) = r_mat_props[ISOTROPIC_ANISOTROPIC_YIELD_RATIO_X];
+        rAs(1, 1) = r_mat_props[ISOTROPIC_ANISOTROPIC_YIELD_RATIO_Y];
+        rAs(2, 2) = r_mat_props[ISOTROPIC_ANISOTROPIC_YIELD_RATIO_XY]; 
+    }
+    for (IndexType i = 0; i < VoigtSize; ++i)
+        rAsInv(i, i) = 1.0 / rAs(i, i);
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<SizeType TVoigtSize>
+void ConstitutiveLawUtilities<TVoigtSize>:: CalculateAnisotropicStrainMapperMatrix(
+    const Matrix& rAnisotropicElasticMatrix,
+    const Matrix& rIsotropicElasticMatrix,
+    const Matrix &rAs,
+    Matrix& rAe
+)
+{
+    if (rAe.size1() != VoigtSize || rAe.size2() != VoigtSize)
+        rAe.resize(VoigtSize, VoigtSize);
+    noalias(rAe) = ZeroMatrix(VoigtSize, VoigtSize);
+
+    Matrix inv_isotropic_elastic_matrix(VoigtSize, VoigtSize);
+    noalias(inv_isotropic_elastic_matrix) = ZeroMatrix(VoigtSize, VoigtSize);
+    double aux_det;
+    MathUtils<double>::InvertMatrix(rIsotropicElasticMatrix, inv_isotropic_elastic_matrix, aux_det);
+    noalias(rAe) = prod(inv_isotropic_elastic_matrix, Matrix(prod(rAs, rAnisotropicElasticMatrix)));
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<>
+void ConstitutiveLawUtilities<6>::CalculateRotationOperatorEuler1(
+    const double EulerAngle1,
+    Matrix& rRotationOperator
+)
+{
+    if (rRotationOperator.size1() != Dimension || rRotationOperator.size2() != Dimension)
+        rRotationOperator.resize(Dimension, Dimension);
+    noalias(rRotationOperator) = ZeroMatrix(Dimension, Dimension);
+
+    const double cos_angle = std::cos(EulerAngle1 * Globals::Pi / 180.0);
+    const double sin_angle = std::sin(EulerAngle1 * Globals::Pi / 180.0);
+
+    rRotationOperator(0, 0) = cos_angle;
+    rRotationOperator(0, 1) = sin_angle;
+    rRotationOperator(1, 0) = -sin_angle;
+    rRotationOperator(1, 1) = cos_angle;
+    rRotationOperator(2, 2) = 1.0;
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<>
+void ConstitutiveLawUtilities<6>::CalculateRotationOperatorEuler2(
+    const double EulerAngle2,
+    Matrix& rRotationOperator
+)
+{
+    if (rRotationOperator.size1() != Dimension || rRotationOperator.size2() != Dimension)
+        rRotationOperator.resize(Dimension, Dimension);
+    noalias(rRotationOperator) = ZeroMatrix(Dimension, Dimension);
+
+    const double cos_angle = std::cos(EulerAngle2 * Globals::Pi / 180.0);
+    const double sin_angle = std::sin(EulerAngle2 * Globals::Pi / 180.0);
+
```

Comment:
There is no singular case with angles?, good

Reply:
As far as I know, the singular cases occur for the inverse problem, which wants to compute the Euler angles from a given rotation matrix. In this case there's no problem.